locals.tf

```diff
@@ -41,4 +41,14 @@ locals {
   sn_public_a = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_a) > 0 ? length(var.public_subnets_a) : 0))
   sn_public_b = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_b) > 0 ? length(var.public_subnets_b) : 0))
   sn_public_c = (local.enable_dynamic_subnets == false ? 1 : (length(var.public_subnets_c) > 0 ? length(var.public_subnets_c) : 0))
+
+  #igw_tags
+  igw_tags =  merge( {  "Name" = var.igw_tags["Name"] }, var.igw_tags) 
+
+  #nat_gw_tags
+  nat_gw_tags = merge({  "Name" = var.nat_gw_tags["Name"] }, var.nat_gw_tags) 
```

Comment:
@eabdelmaqsod 
nat_gw_tags = merge({"Name" = var.nat_gw_name}, var.nat_gw_tags)

+ declare "var.nat_gw_name" (default = aws_vpc.main.id)

Reply:
@Geartrixy ,

There is no need to specify a new var for name to just set default value, I set the local value to vpc_name if the no tags are passed